vesin/include/vesin_cuda.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef VESIN_CUDA_HPP
+#define VESIN_CUDA_HPP
+
+#include <vector>
+
+#include "vesin.h"
+
+namespace vesin {
+namespace cuda {
+
+#ifndef VESIN_CUDA_MAX_PAIRS_PER_POINT
+/// @brief Default maximum number of pairs per point on the GPU (can be
+/// overridden).
+#define VESIN_CUDA_MAX_PAIRS_PER_POINT 512
+#endif
+
+struct CudaNeighborListExtras {
+    unsigned long* length_ptr = nullptr; // GPU-side counter
+    unsigned long capacity = 0;          // Current capacity per device
+    int* cell_check_ptr = nullptr;       // GPU-side status code for checking cell
+    int allocated_device = -1;           // which device are we currently allocated on
+
+    ~CudaNeighborListExtras() {
+        // cleanup handled in `free_neighbors`
```

Comment:
the things to cleanup are `length_ptr` and `cell_check_ptr`, right? I don't see anything done to them in `free_neighbors`.

In general it would be nice if the cleanup could happen here instead of a separate function, the destructor will be called by `delete` in `free_neighbors`.